epochStart/bootstrap/simpleMetaBlockInterceptor.go

```diff
@@ -0,0 +1,109 @@
+package bootstrap
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+)
+
+const timeToWaitBeforeCheckingReceivedMetaBlocks = 500 * time.Millisecond
+const numTriesUntilExit = 5
+
+type simpleMetaBlockInterceptor struct {
+	marshalizer            marshal.Marshalizer
+	hasher                 hashing.Hasher
+	mutReceivedMetaBlocks  sync.RWMutex
+	mapReceivedMetaBlocks  map[string]*block.MetaBlock
+	mapMetaBlocksFromPeers map[string][]p2p.PeerID
+}
+
+// NewSimpleMetaBlockInterceptor will return a new instance of simpleMetaBlockInterceptor
+func NewSimpleMetaBlockInterceptor(marshalizer marshal.Marshalizer, hasher hashing.Hasher) (*simpleMetaBlockInterceptor, error) {
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+	if check.IfNil(hasher) {
+		return nil, ErrNilHasher
+	}
+
+	return &simpleMetaBlockInterceptor{
+		marshalizer:            marshalizer,
+		hasher:                 hasher,
+		mutReceivedMetaBlocks:  sync.RWMutex{},
+		mapReceivedMetaBlocks:  make(map[string]*block.MetaBlock),
+		mapMetaBlocksFromPeers: make(map[string][]p2p.PeerID),
+	}, nil
+}
+
+// ProcessReceivedMessage will receive the metablocks and will add them to the maps
+func (s *simpleMetaBlockInterceptor) ProcessReceivedMessage(message p2p.MessageP2P, _ func(buffToSend []byte)) error {
+	var mb block.MetaBlock
+	err := s.marshalizer.Unmarshal(&mb, message.Data())
+	if err == nil {
```

Comment:
if err != nil return err. After that the rest of the code.

Reply:
done